src/post_process/checks.rs

```diff
@@ -178,20 +197,40 @@ fn find_system_libs(output: &Output) -> Result<GlobSet, globset::Error> {
         return system_libs.build();
     }
 
-    if let Some(sysroot_package) = output
+    // Try to locate a `sysroot_<platform>` package in either the regular build
+    // dependencies *or* the cache build dependencies. The first one found is
+    // used to derive the list of system libraries.
+    let mut sysroot_package = output
         .finalized_dependencies
-        .clone()
-        .expect("failed to get the finalized dependencies")
-        .build
+        .as_ref()
+        .and_then(|deps| deps.build.as_ref())
         .and_then(|deps| {
-            deps.resolved.into_iter().find(|v| {
+            deps.resolved.iter().find(|v| {
                 v.file_name.starts_with(&format!(
                     "sysroot_{}",
                     output.build_configuration.target_platform
                 ))
             })
         })
-    {
+        .cloned();
+
+    if sysroot_package.is_none() {
```

Comment:
Instead of writing everything twice, can we merge the finalized dependencies into one object?